packages/components/src/range-control/index.tsx

```diff
@@ -40,6 +40,15 @@ import type { WordPressComponentProps } from '../ui/context';
 
 const noop = () => {};
 
+function useUniqueId( idProp?: string ) {
+	const id = useInstanceId(
+		UnforwardedRangeControl,
+		'inspector-range-control'
+	);
+
+	return idProp || id;
```

Comment:
Would it be better to allow for empty string as the value of `isProp` ?

```suggestion
	return idProp ?? id;
```

Reply:
> Would it be better to allow for empty string as the value of `idProp` ?

I'm not sure as I can think of a scenario where an empty string would be an acceptable value. Currently, the range control auto-generates its own ID, and always uses it. The change just means that it can be overridden, and if the range control expects a value for any reason, an empty string might cause confusion.

But this is not a strongly held opinion! I'd be happy with your suggested change if those more familiar with the code than me think it appropriate. I took it from a current pattern that exists in [a](https://github.com/WordPress/gutenberg/blob/9189c3a34714352b92d19a3ef33149b76d8efe64/packages/components/src/input-control/input-base.tsx#L27-L32) [few](https://github.com/WordPress/gutenberg/blob/9189c3a34714352b92d19a3ef33149b76d8efe64/packages/components/src/input-control/index.tsx#L25-L30) [places](https://github.com/WordPress/gutenberg/blob/9189c3a34714352b92d19a3ef33149b76d8efe64/packages/components/src/select-control/index.tsx#L25-L30) already (to the point that it might be worth adding a `useOverridableInstanceId` function in `@wordpress/compose`!).

Reply:
Thank you, this makes sense to me!